src/root/posts.js

```diff
@@ -0,0 +1,263 @@
+import m from 'mithril';
+import postgrest from 'mithril-postgrest';
+import _ from 'underscore';
+import I18n from 'i18n-js';
+import h from '../h';
+import models from '../models';
+import projectDashboardMenu from '../c/project-dashboard-menu';
+import postsPreview from '../c/posts-preview';
+import rewardVM from '../vms/reward-vm';
+
+const posts = {
+    controller(args) {
+
+        let deleteFormSubmit;
+        const showPreview = m.prop(false),
+            projectPosts = m.prop(),
+            fields = {
+                title: m.prop(''),
+                comment_html: m.prop(''),
+                reward_id: m.prop('-1')
+            },
+            filterVM = postgrest.filtersVM({
+                project_id: 'eq'
+            }),
+            togglePreview = () => {
+                showPreview(true);
+            },
+            project_id = args.root.getAttribute('data-id'),
+            projectDetails = m.prop([]),
+            loader = postgrest.loaderWithToken,
+            setProjectId = () => {
+                try {
+                    const project_id = m.route.param('project_id');
+
+                    filterVM.project_id(project_id);
+                } catch (e) {
+                    filterVM.project_id(args.root.getAttribute('data-id'));
+                }
+            },
+            showRecipientes = (post) => {
+                if (post.recipients == 'public') {
+                    return 'Todo mundo (apoiadores e não apoiadores)';
+                } else if (post.recipients == 'backers') {
+                    return 'Todos os apoiadores';
+                } else {
+                    let reward = _.find(rewardVM.rewards(), (reward) => {
```

Comment:
if we are only returning we may skip the function block: `reward => reward.id == post.reward_id`